src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs

```diff
@@ -292,6 +295,26 @@ public override void Dispose()
 
         #region private and internal methods
 
+        /// <summary>
+        /// Notifies all relevant Dynamo features (UI elements, commands) that the Graph exection has been enabled/disabled. 
+        /// </summary>
+        void NotifyOfGraphRunChanged()
+        {
+            RaisePropertyChanged("RunButtonEnabled");
+            RaisePropertyChanged("RunButtonToolTip");
+
+            if (Application.Current != null)
+            {
+                Application.Current.Dispatcher.Invoke(new Action(() =>
```

Comment:
any reason to use invoke instead of begin invoke?

Reply:
This code was moved (not added). 
I am not sure what the implications would be if we made the change